server/routes/cohorts.js

```diff
@@ -57,4 +62,63 @@ router.get(
   })
 );
 
+// Assign participant
+router.post(
+  '/:id/assign/:participantId',
+  [applyMiddleware(keycloak.allowRolesMiddleware('health_authority', 'employer'))],
+  asyncMiddleware(async (req, res) => {
+    const { user_id: userId, sub: localUserId } = req.user;
+    const user = userId || localUserId;
+    const { id, participantId } = req.params;
+    if (id && participantId) {
+      const [participant] = await getParticipantByID({ id: +participantId });
+      if (!participant) {
+        return res.status(400).send('Invalid participant id');
+      }
+      const [cohort] = await getCohort(+id);
+      if (!cohort) {
+        return res.status(400).send('Invalid cohort id');
+      }
+      const response = await assignCohort({ id: cohort.id, participantId: participant.id });
+      logger.info({
+        action: 'cohort_participant_assign',
+        performed_by: {
+          user,
+        },
+        cohortId: cohort.id || '',
+        participantId: participant.id,
+      });
+      return res.status(201).json(response);
+    }
+    return res.status(400).send('Cohort id and participant id required');
+  })
+);
+
+// Get Assigned cohort for participant
+router.get(
+  '/assigned-participant/:id',
+  asyncMiddleware(async (req, res) => {
+    const { user_id: userId, sub: localUserId } = req.user;
+    const user = userId || localUserId;
+    const { id } = req.params;
+    if (!id) {
+      return res.status(400).send('Participant id required');
+    }
+    const [participant] = await getParticipantByID({ id: +id });
+    if (!participant) {
+      return res.status(400).send('Invalid participant id');
+    }
+    const [cohort] = (await getAssignCohort({ participantId: participant.id })) || [{}];
+    logger.info({
+      action: 'cohort_participant_get_assigned',
+      performed_by: {
+        user,
+      },
+      cohortId: cohort.id || '',
```

Comment:
is it possible that `cohort.id` could be undefined here?

Reply:
Yes, if no cohort assigned ....